### PR TITLE
Continue printing out redirected process output after we detect browser launch URL.

### LINKED ISSUE #44259: Dotnet watch pops browser and asks for auth token, but token is never printed to the console
See https://github.com/dotnet/aspire/issues/5010

Image

cc @tmat @phil-allen-msft

### PR DESCRIPTION
**Pull request by tmat**
The logic that scans for browser launch URL in the process output stopped forwarding the output after the match has been found. This removes information logged by the web app after that point.

Fixes regression from .NET 8.0.

### CODE REVIEW

**Comment by tmat** [comment]
@davidfowl @eerhardt

File: test/dotnet-watch.Tests/Watch/BrowserLaunchTests.cs [hunk 1]
```diff
@@ -22,8 +22,11 @@ public async Task LaunchesBrowserOnStart()
 
             App.Start(testAsset, [], testFlags: TestFlags.MockBrowser);
 
+            // check that all app output is printed out:
+            await App.AssertOutputLine(line => line.Contains("Content root path:"));
+
             // Verify we launched the browser.
-            await App.AssertOutputLineStartsWith("dotnet watch ⌚ Launching browser: https://localhost:5001/");
+            Assert.Contains(App.Process.Output, line => line.Contains("Launching browser: https://localhost:5001/"));
```
Comment (davidfowl) [inline test/dotnet-watch.Tests/Watch/BrowserLaunchTests.cs:29]: Can we add an assert for the Ctrl+C output as well? That comes after this.

Comment (am11) [reply test/dotnet-watch.Tests/Watch/BrowserLaunchTests.cs:29]: while preserving the emojis please: `dotnet watch ⌚ keeping it fresh while we code in style! 😎`

**Comment by marcpopMSFT** [comment]
@tmat is this expected for GA and how close is it? I'm checking in on all remaining active 9.0.1xx PRs

**Comment by davidfowl** [comment]
Yes this is a regression

**Comment by tmat** [comment]
@marcpopMSFT Yes, for GA. Some dotnet new integration tests are intermittently failing in the CI. Restarting and hoping to merge today.

**Comment by tmat** [comment]
@marcpopMSFT Source build is failing with

```
Microsoft.DotNet.SourceBuild.SmokeTests.SourceBuiltArtifactsTests.VerifyVersionFile

Assert.Equal() Failure: Strings differ
                  ↓ (pos 7)
Expected: "9.0.100-rtm.24518.1"
Actual:   "9.0.100"
```

Unrelated to this PR.

**Review by davidfowl — APPROVED** [review]

**Pull request merged**